web/src/topic_list_data.js

```diff
@@ -107,17 +96,50 @@ export function get_list_info(stream_id, zoomed, search_term) {
             is_zero: num_unread === 0,
             is_muted: is_topic_muted,
             is_active_topic,
-            url: hash_util.by_stream_topic_url(stream_id, topic_name),
+            url: hash_util.by_stream_topic_url(stream_info.stream_id, topic_name),
             contains_unread_mention,
         };
 
         items.push(topic_info);
     }
+}
+
+export function get_list_info(stream_id, zoomed, search_term) {
+    const stream_information = {
```

Comment:
`stream_info` is a questionable name for this -- it's not really information about the stream. I'm not sure what a great name is, though. 

(As a side note, you should almost never have a thing you call `stream_information` in one place and `stream_info` in the other -- it's way way easier to read code if a variable/value doesn't get randomly renamed between different abbreviations across a code path. I think I've pointed this out in your work before -- please pay attention to details like this! Whenever you've written code that you think works, you should do another read of it looking at what you called things and make sure it feels good).

Reply:
I think my proposal is that you make `stream_id` not part of this data structure -- it can be just the first parameter to `choose_topics` -- and `items` could be added to this data structure rather than an independent parameter -- and then this can be `topic_choice_state`.